### PR TITLE
feat(task): Add task types

### PR DESCRIPTION
**Pull request by lyondhill**
Add a type to task's to allow tasks to be filterable.

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)

### CODE REVIEW

File: http/swagger.yml [hunk 1]
```diff
@@ -6551,6 +6551,9 @@ components:
         id:
           readOnly: true
           type: string
+        type:
+          description: The type of task, this can be used for filtering
+          type: string
```
Comment (chnn) [inline http/swagger.yml]: Is there a fixed set of values that this `type` can be? If so, it would be helpful to add an `enum` field here.

Comment (lyondhill) [reply http/swagger.yml]: Anything except the wildcard "*" no limits currently.

**Comment by lyondhill** [comment]
@gshif We are currently not limiting task types (other than using a "wildcard" as the task type on create). I can make the description in the swagger more descriptive.

**Comment by gshif** [comment]
That would be great

File: kv/task.go [hunk 2]
```diff
@@ -296,6 +304,14 @@ func (s *Service) findTaskByOrg(ctx context.Context, tx Tx, filter influxdb.Task
...
+				ft := ""
+				filter.Type = &ft
+			}
+			if *filter.Type != influxdb.TaskTypeWildcard && *filter.Type != t.Type {
+				t = nil
+			}
+
 			// insert the new task into the list
 			if t != nil {
 				ts = append(ts, t)
```
Comment (GeorgeMac) [inline kv/task.go]: Take it or leave it suggestion. I found the flow a little hard to parse and a bit concerned about all the pointers. I would suggest something like:
```go
if t != nil {
  typ := ""
  if *filter.Type != nil {
    typ = *filter.Type
  }

  // if the filter type matches task type or filter type is a wildcard
  if typ == t.Type  || typ == influxdb.TakeTypeWildcard {
    ts = append(ts, t)
  }
}
```
Actually it appears in the original code that `t` could be nil. At least that nil check guard suggests it could be. So perhaps you do need a nil check on t somewhere first, otherwise the `t.Type` could nil pointer dereference. (either that or t is never really nil which is actually more than likely).

File: task/servicetest/servicetest.go [hunk 3]
```diff
@@ -1437,3 +1443,87 @@ option task = {
...
+	}
+
+	// get all tasks
+	wc := influxdb.TaskTypeWildcard
+	tasks, _, err = sys.TaskService.FindTasks(sys.Ctx, influxdb.TaskFilter{OrganizationID: &cr.OrgID, Type: &wc})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// this is to account for the concurrency test creating tasks in the same organization
```
Comment (GeorgeMac) [inline task/servicetest/servicetest.go]: Another take it or leave it:

If you want to get precise here you could update the CredsFunc to take a *testing.T here: https://github.com/influxdata/influxdb/blob/c4fd4d79865873795b02dbdd1bcf26b3fd3e6fcd/http/task_test.go#L70

And then move org creation into the creds func call as well.
Then you can properly scope the organisation to the subtest name. That way you don't share organisations between parallel test runs leading to unpredictable results. Could avoid someone in the future deleting that concurrency test and breaking this one for example.
Also this looks like only happens in the http/task_test.go as that defines its own CredFunc override which causes this problem.

Comment (lyondhill) [reply task/servicetest/servicetest.go]: woo fresh eyes are so nice. Thanks! that little change will be very good for this testing suite.

File: kv/task.go (hunk 2 above)
Comment (lyondhill) [reply kv/task.go]: good find. the call above `influxdb.IDFromString(string(v))` used to be able to return a nil task with no error, I am not sure it does that anymore. I do like the logic in your solution better, its more clean.
